lib/client.js

```diff
@@ -874,18 +993,55 @@ MqttClient.prototype._handlePingresp = function () {
  */
 
 MqttClient.prototype._handleConnack = function (packet) {
-  var rc = packet.returnCode
-  var errors = [
-    '',
-    'Unacceptable protocol version',
-    'Identifier rejected',
-    'Server unavailable',
-    'Bad username or password',
-    'Not authorized'
-  ]
+  var version = this.options.protocolVersion
+  var rc = version === 5 ? packet.reasonCode : packet.returnCode
+  var errors = {
+    0: '',
+    1: 'Unacceptable protocol version',
+    2: 'Identifier rejected',
+    3: 'Server unavailable',
+    4: 'Bad username or password',
+    5: 'Not authorized',
+    128: 'Unspecified error',
+    129: 'Malformed Packet',
+    130: 'Protocol Error',
+    131: 'Implementation specific error',
+    132: 'Unsupported Protocol Version',
+    133: 'Client Identifier not valid',
+    134: 'Bad User Name or Password',
+    135: 'Not authorized',
+    136: 'Server unavailable',
+    137: 'Server busy',
+    138: 'Banned',
+    140: 'Bad authentication method',
+    144: 'Topic Name invalid',
+    149: 'Packet too large',
+    151: 'Quota exceeded',
+    153: 'Payload format invalid',
+    154: 'Retain not supported',
+    155: 'QoS not supported',
+    156: 'Use another server',
+    157: 'Server moved',
+    159: 'Connection rate exceeded'
+  }
```

Comment:
this should really go to the top of the file. There is no need to declare this here.

Reply:
resolve